doc/Data Plane Code Generation/DeveloperDrivenEvolution.md

```diff
@@ -0,0 +1,199 @@
+Convenience API is an additive layer on top of protocol mathod so it could produce improved user experience.
+
+To add a convenience API that takes and returns strongly-typed input and output models instead of raw JSON, steps are:
+1. [Create the convenience method with new signature](#create-the-convenience-method-with-new-signature)
+- [Replace input parameters](#replace-input-parameters)
+- [Replace output value](#replace-output-value)
+- [Rename method if needed](#rename-method-if-needed)
+2. [Create the models](#create-the-models)
+3. [Implement the method](#implement-the-method)
+- [Implement GET method](#implement-get-method)
+- [Implement POST method](#implement-post-method)
+- [Implement GET paging method](#implement-get-paging-method)
+
+## Create the convenience method with new signature
+To create a convenience method on top of the protocol method, you should not modify generated code. The convenience methods should be added in a saparate file as partial class of the class containing protocol methods. You could check out the full examples in [Implement the method](#implement-the-method).
+
+Below is how you could create the convenience signature from protocol method. Here is an example of your generated protocol method.
+```C#
+public virtual async Task<Response> CreateMetricFeedbackAsync(RequestContent content, RequestContext context = null);
+```
+You should do the following steps to create the convenience method signature:
+### Replace input parameters
+- Replace the parameter `RequestContext` with an optional `CancellationToken` with default value.
+- If there is an input parameter `RequestContent`, replace it with a wanted input type (e.g., `MetricFeedback`).
+
+After these steps, your convenience method signature will be like:
+```C#
+public virtual async Task<Response> CreateMetricFeedbackAsync(MetricFeedback feedback, CancellationToken cancellationToken = default);
+```
+### Replace output value
+- Replace the return value `Response` with `Response<T>` (`T` is your wanted output type, e.g., `MetricFeedback`).
+
+After these steps, your convenience method signature will be like:
+```C#
+public virtual async Task<Response<MetricFeedback>> CreateMetricFeedbackAsync(MetricFeedback feedback, CancellationToken cancellationToken = default);
+```
+### Rename method if needed
+Renaming method should follow the rules:
+
+**It is needed if ambiguous call exists**
+
+This situation happens when your protocol method and convenience method have the same required paramater list. E.g.,
+```C#
+// protocol method
+public virtual async Task<Response> GetMetricFeedbackAsync(string feedId, RequestContext context = null);
+// Convenience method
+public virtual async Task<Response<MetricFeedback>> GetMetricFeedbackAsync(string feedId, CancellationToken cancellationToken = default);
+```
+Ambiguous call exists when you call
+```C#
+GetMetricFeedbackAsync(feedId);
+```
+In this situation, you have to change the convenience name.
```

Comment:
Do we have a rule about what is the acceptable name for the 'convenience name'?
In this case, how can i rename 'GetMetricFeedbackAsync'